### PR TITLE
Fix Clock Propagation

### PR DESCRIPTION
**Pull request by donn**
* `OpenROAD.STA*PNR`
  * New environment variable made accessible to SDC files used during Multi-Corner STA steps, `OPENLANE_SDC_IDEAL_CLOCKS`, set to `1` for pre-PnR. Band-aid until the SDC situation is properly discussed and addressed (in a potentially breaking change.)
  * Fixed issue where the clock was always propagated after `STAPrePNR` regardless the information in the SDC file.
  * For backwards compatibility, `STAPrePNR` unsets all propagated clocks and the rest set all propagated clocks IF the SDC file lacks the strings `set_propagated_clock` or `unset_propagated_clock`.

---
Depends on https://github.com/efabless/openlane2-step-unit-tests/pull/35
Depends on https://github.com/efabless/openlane2-ci-designs/pull/32

### CODE REVIEW

**Review by kareefardi — CHANGES_REQUESTED** [review]
`OPENLANE_SDC_IDEAL_CLOCKS` is an internal variable. It should be prefixed with `_` (or `__`?)

**Comment by donn** [comment]
It is not an internal variable. It is a variable exposed to SDC files.

What I can do is add it to the step documentation.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ body:
         You also need to upload some files so we can reproduce your issue.
 
         If a step fails, you can create a reproducible by going
-        into the step directory, then typing `openlane.steps create-reproducible -d .`
+        into the step directory, then typing `openlane.steps create-reproducible`
```
[the remark this answers is not in the document]
Comment (donn) [reply .github/ISSUE_TEMPLATE/bug_report.yml:47]: no

**Comment by kareefardi** [comment]
> It is not an internal variable. It is a variable exposed to SDC files.
> 
> What I can do is add it to the step documentation.

Isn't it forced to specific value in some steps?

**Review by kareefardi — APPROVED** [review]

**Comment by kareefardi** [comment]
I think users may miss `OPENLANE_SDC_IDEAL_CLOCKS` but since it is getting removed in OpenLane 3 perhaps it is not that big of a deal.

**Review by kareefardi — APPROVED** [review]

**Pull request merged**